### PR TITLE
[dotnet] Include the marshal-ilgen component. Fixes #16836.

### LINKED ISSUE #16836: marshal-ilgen component must be included
After reverting, marshal-ilgen has been moved to again component: https://github.com/dotnet/runtime/pull/75542. This is a step in eventually allowing the component to be dropped when not needed (reducing the runtime size by around 100kb). For now, it must be available to the runtime. So it needs to be added here before this change is consumed by xamarin-macios.

@dalexsoto @rolfbjarne

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes https://github.com/xamarin/xamarin-macios/issues/16836.
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1751368.

### CODE REVIEW

**Comment by mauroa** [comment]
@rolfbjarne will this get included for the Hot Restart pre-built app or should we add the same to the pre-built .csproj?

File: dotnet/targets/Xamarin.Shared.Sdk.targets [hunk 1]
```diff
@@ -395,6 +395,7 @@
 			<_MonoComponent Include="hot_reload" Condition="'$(MtouchInterpreter)' == 'true'" />
 			<_MonoComponent Include="debugger" Condition="'$(_BundlerDebug)' == 'true'" />
 			<_MonoComponent Include="diagnostics_tracing" Condition="'$(_BundlerDebug)' == 'true'" />
+			<_MonoComponent Include="marshal-ilgen" Condition="'$(_AppleExcludeMarshalIlgenComponent)' != 'true'" />
```
Comment (mauroa) [inline dotnet/targets/Xamarin.Shared.Sdk.targets:398]: Will this get included for the Hot Restart pre-built app or should we add the same to the pre-built .csproj?

**Review by mandel-macaque — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated (https://github.com/xamarin/maccore/issues/1265).

File: dotnet/targets/Xamarin.Shared.Sdk.targets (hunk 1 above)
Comment (rolfbjarne) [reply dotnet/targets/Xamarin.Shared.Sdk.targets:400]: It will get included automatically.

**Pull request merged**